employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/gwtjackson/ServerSideExceptionDeserializer.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.gwtui.client.gwtjackson;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Singleton;
+
+import com.github.nmorel.gwtjackson.client.stream.JsonReader;
+import com.github.nmorel.gwtjackson.client.stream.JsonToken;
+import com.github.nmorel.gwtjackson.client.stream.impl.DefaultJsonReader;
+import com.github.nmorel.gwtjackson.client.stream.impl.StringReader;
+import org.optaweb.employeerostering.shared.exception.ServerSideExceptionInfo;
+import org.optaweb.employeerostering.shared.exception.ServerSideExceptionInfo.ServerSideExceptionInfoFields;
+
+@Singleton
+public class ServerSideExceptionDeserializer {
```

Comment:
This class has not been covered by any tests - please see the SonarCloud report.